### PR TITLE
Mini Project completed.

### PR DESCRIPTION
**Pull request by hrishikeshmane**
## Features
* Search
* Archive/Unarchive notes
* Grid/List view
* Login in with Google
* Persistent database
* Colors and Tags feature
* Filter by tags and colors (useReducer/useContext)
* Sort by date

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,7 +1,102 @@
...
+    try {
+      await setDoc(
+        doc(db, "Notes", sessionStorage.getItem("email")),
+        {
+          username: sessionStorage.getItem("email"),
+        },
+        { merge: true }
+      );
+    } catch (e) {
+      console.error("Error adding document: ", e);
```
Comment (Aditi0505) [inline src/App.js:36]: Display error on UI

File: src/Filter.jsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      open={openFilter}
+      onClose={() => setOpenFilter((prevState) => !prevState)}
+    >
+      <Box
+        sx={{
+          p: 2,
+          backgroundColor: "#262729",
+          height: "100%",
+          width: 250,
+          color: "white",
```
Comment (Aditi0505) [inline src/Filter.jsx]: Use hex code here as well

File: src/InputBox.css [hunk 3]
```diff
@@ -0,0 +1,120 @@
+::-webkit-scrollbar {
+  display: none;
+}
+
+.input-container {
+  width: 95%;
+  max-width: 600px;
```
Comment (Aditi0505) [inline src/InputBox.css]: Prefer rem over px. Use px only when value <= 15.

File: src/InputBox.css [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+}
+input[type="text"]:focus-visible {
+  outline: none !important  ;
+}
+textarea {
+  width: 100% !important;
+  height: 100px !important;
+
+  font-family: Roboto, Arial, sans-serif;
+  font-size: 0.875rem !important;
```
Comment (Aditi0505) [inline src/InputBox.css]: Avoid using important at so many places. Rather create a more specific class and then apply it.

File: src/context/filter-reducer.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+const filterReducer = (filterState, action) => {
```
Comment (Aditi0505) [inline src/context/filter-reducer.js:1]: Can destructure action here.

**Review by Aditi0505 — APPROVED** [review]
Left few suggestions for you to look at.
- Good folder structure and clean coding
- Good naming of components

File: src/App.css [hunk 6]
```diff
@@ -6,3 +6,44 @@
...
+  margin: auto;
+  border-radius: 10px;
+  padding: 1rem 4rem;
+  justify-content: center;
+  align-items: center;
+  border: none;
+  cursor: pointer;
+}
+.signin-button:hover {
+  background-color: rgb(169, 200, 228);
```
Comment (SanketDhabarde) [inline src/App.css]: Be consistent with colors. either use hex or RGB. And it will be a good practice to add in variables

**Review by SanketDhabarde — APPROVED** [review]
### Left one suggestion
Also,
- You can create the components folder
### Everything else looks good
- Like that you used firebase
- Properly indented code
- Looks clean and understandable code

**Pull request merged**